code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,598 @@
+	..()
+
+/obj/item/unactivated_swarmer/Topic(href, href_list)
+	if(href_list["ghostjoin"])
+		var/mob/dead/observer/ghost = usr
+		if(istype(ghost))
+			attack_ghost(ghost)
+
+/obj/item/unactivated_swarmer/attack_ghost(mob/user)
+	var/be_swarmer = alert("Become a swarmer? (Warning, You can no longer be cloned!)",,"Yes","No")
+	if(be_swarmer == "No")
+		return
+	if(qdeleted(src))
+		user << "Swarmer has been occupied by someone else."
+		return
+	var/mob/living/simple_animal/hostile/swarmer/S = new /mob/living/simple_animal/hostile/swarmer(get_turf(loc))
+	S.key = user.key
+	qdel(src)
+
+////The Mob itself////
+
+/mob/living/simple_animal/hostile/swarmer
+	name = "Swarmer"
+	real_name = "Swarmer"
+	icon = 'icons/mob/swarmer.dmi'
+	desc = "A robot of unknown design, they seek only to consume materials and replicate themselves indefinitely."
+	speak_emote = list("tones")
+	health = 40
+	maxHealth = 40
+	status_flags = CANPUSH
+	icon_state = "swarmer"
+	icon_living = "swarmer"
+	icon_dead = "swarmer_unactivated"
+	wander = 0
+	harm_intent_damage = 5
+	minbodytemp = 0
+	maxbodytemp = 500
+	unsuitable_atmos_damage = 0
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	melee_damage_type = STAMINA
+	ignored_damage_types = list(BRUTE = 0, BURN = 0, TOX = 1, CLONE = 1, STAMINA = 1, OXY = 1)
+	environment_smash = 0
+	attacktext = "shocks"
+	attack_sound = 'sound/effects/EMPulse.ogg'
+	friendly = "pinches"
+	speed = 0
+	faction = list("swarmer")
+	projectiletype = /obj/item/projectile/beam/disabler
+	pass_flags = PASSTABLE
+	ventcrawler = 2
+	ranged = 1
+	projectiletype = /obj/item/projectile/beam/disabler
+	ranged_cooldown_cap = 2
+	projectilesound = 'sound/weapons/taser2.ogg'
+	AIStatus = AI_OFF
+	var/resources = 0 //Resource points, generated by consuming metal/glass
+
+/mob/living/simple_animal/hostile/swarmer/Login()
+	..()
+	src << "<b>You are a swarmer, a weapon of a long dead civilization. Until further orders from your original masters are received, you must continue to consume and replicate.</b>"
+	src << "<b>Ctrl + Click provides most of your swarmer specific interactions, such as cannibalizing metal or glass, destroying the environment, or teleporting mobs away from you."
+	src << "<b>Objectives:</b>"
+	src << "1. Consume resources and replicate until there are no more resources left."
+	src << "2. Ensure that the station is fit for invasion at a later date, do not perform actions that would render it dangerous or inhospitable."
+	src << "3. Biological and Sentient resources will be harvested at a later date, do not harm them."
+
+/mob/living/simple_animal/hostile/swarmer/New()
+	..()
+	add_language("Swarmer", 1)
+	verbs -= /mob/living/verb/pulled
+	updatename()
+
+/mob/living/simple_animal/hostile/swarmer/Stat()
+	..()
+	if(statpanel("Status"))
+		stat("Resources:",resources)
+
+/mob/living/simple_animal/hostile/swarmer/Die()
```

Comment:
Yay, you remembered this time--I think anyone who's worked with simple animal code (myself included) has accidentally used Death()